rdkit/Chem/UnitTestSurf.py

```diff
@@ -94,47 +97,158 @@ def testHsAndTPSA(self):
+        val = func(data.mol)
+      except Exception:
+        val = 666
+      self.assertAlmostEqual(
+        val, data.expected, delta=1e-4,
+        msg='line {0.lineNo}, mol {0.smiles} (calc = {1}) should have val = {0.expected}'.format(
+          data, val))
+
+  def testLabuteASALong(self):
+    if not doLong:
+      raise unittest.SkipTest('long test')
+    col = 6
+    self.__testDesc('PP_descrs_regress.csv', col, lambda x: MolSurf.LabuteASA(x, includeHs=1))
+    self.__testDesc('PP_descrs_regress.2.csv', col, lambda x: MolSurf.LabuteASA(x, includeHs=1))
+
+  def testTPSALong(self):
+    col = 28
+    self.__testDesc('PP_descrs_regress.csv', col, MolSurf.TPSA)
     if doLong:
-      inName = os.path.join(RDConfig.RDCodeDir, 'Chem', 'test_data', 'NCI_5K_TPSA.csv')
-      with open(inName, 'r') as inF:
-        lines = inF.readlines()
-      for line in lines:
-        if line[0] != '#':
-          line.strip()
-          smi, ans = line.split(',')
-          ans = float(ans)
-          mol = Chem.MolFromSmiles(smi)
-          mol = Chem.AddHs(mol)
-          calc = MolSurf.TPSA(mol)
-          self.assertTrue(
-            feq(calc, ans), 'bad TPSA for SMILES %s (%.2f != %.2f)' % (smi, calc, ans))
-
-
-if __name__ == '__main__':
-  import sys, getopt, re
-  doLong = 0
-  if len(sys.argv) > 1:
-    args, extras = getopt.getopt(sys.argv[1:], 'l')
-    for arg, val in args:
-      if arg == '-l':
-        doLong = 1
-      sys.argv.remove('-l')
-  if doLong:
-    for methName in dir(TestCase):
-      if re.match('_test', methName):
-        newName = re.sub('_test', 'test', methName)
-        exec('TestCase.%s = TestCase.%s' % (newName, methName))
-
+      self.__testDesc('PP_descrs_regress.2.csv', col, MolSurf.TPSA)
+
+  def testMOELong(self):
+    if not doLong:
+      raise unittest.SkipTest('long test')
+    fName = 'PP_descrs_regress.VSA.csv'
+    self.__testDesc(fName, 1, MolSurf.SMR_VSA1)
+    self.__testDesc(fName, 2, MolSurf.SMR_VSA10)
+    self.__testDesc(fName, 3, MolSurf.SMR_VSA2)
+    self.__testDesc(fName, 4, MolSurf.SMR_VSA3)
+    self.__testDesc(fName, 5, MolSurf.SMR_VSA4)
+    self.__testDesc(fName, 6, MolSurf.SMR_VSA5)
+    self.__testDesc(fName, 7, MolSurf.SMR_VSA6)
+    self.__testDesc(fName, 8, MolSurf.SMR_VSA7)
+    self.__testDesc(fName, 9, MolSurf.SMR_VSA8)
+    self.__testDesc(fName, 10, MolSurf.SMR_VSA9)
+    self.__testDesc(fName, 11, MolSurf.SlogP_VSA1)
+    self.__testDesc(fName, 12, MolSurf.SlogP_VSA10)
+    self.__testDesc(fName, 13, MolSurf.SlogP_VSA11)
+    self.__testDesc(fName, 14, MolSurf.SlogP_VSA12)
+
+    fName = 'PP_descrs_regress.VSA.2.csv'
+    self.__testDesc(fName, 1, MolSurf.SMR_VSA1)
+    self.__testDesc(fName, 2, MolSurf.SMR_VSA10)
+    self.__testDesc(fName, 11, MolSurf.SlogP_VSA1)
+    self.__testDesc(fName, 12, MolSurf.SlogP_VSA10)
+    self.__testDesc(fName, 13, MolSurf.SlogP_VSA11)
+    self.__testDesc(fName, 14, MolSurf.SlogP_VSA12)
+
+
+class TestCase_python(unittest.TestCase):
```

Comment:
General comment: I think the python implementations of these descriptors should probably just be removed. They are really just historical artifacts and shouldn't be used in client code. As long as they exist, it makes sense to have test for them though, so thanks for doing this.